hunav_evaluator_node:
  ros__parameters:

    # Specify the frecuency of data capture (Hz).
    # value 0.0 indicates that data is captured as the same 
    # freq than data is published.
    frequency: 1.0

    # optional tag that can be used to identify among 
    # different scenarios or goal trajectories
    experiment_tag: '1'

    # A .txt file will be created with this name (indicate the full path).
    # If the file already exists, the file will be open
    # and a new line of results will be added at the end
    # of the file. 
    result_file: 'metrics.txt'


    # Boolean to indicate whether the data recording must wait to 
    # receive a goal message (geometry_msgs/msg/PoseStamped) in the 
    # topic */goal_pose* to start or not. This way, the recording 
    # can be coordinated with the start of the robot navigation.
    use_navgoal_to_start: true


    # The list of metrics to be computed.
    metrics:
    
      # Pérez-Higueras, N., Caballero, F. & Merino, L. 
      # Teaching Robot Navigation Behaviors to Optimal RRT Planners. 
      # Int.J. of Soc. Robotics 10, 235–249 (2018). https://doi.org/10.1007/s12369-017-0448-1
      time_to_reach_goal: true
      path_length: true
      cumulative_heading_changes: true
      avg_distance_to_closest_person: true
      minimum_distance_to_people: true
      intimate_space_intrusions: true
      personal_space_intrusions: true
      social_space_intrusions: true
      group_intimate_space_intrusions: true
      group_personal_space_intrusions: true
      group_social_space_intrusions: true
      
      
      # N. Tsoi et al., "SEAN 2.0: Formalizing and Generating Social Situations
      # for Robot Navigation," in IEEE Robotics and Automation Letters, vol. 7,
      # no. 4, pp. 11047-11054, Oct. 2022, doi: 10.1109/LRA.2022.3196783.
      completed: true
      minimum_distance_to_target: true
      final_distance_to_target: true
      robot_on_person_collision: true
      person_on_robot_collision: true
      time_not_moving: true
      
      # SocNavBench: A Grounded Simulation Testing Framework for Evaluating Social Navigation
      #ABHIJAT BISWAS, ALLAN WANG, GUSTAVO SILVERA, AARON STEINFELD, and HENNY AD-MONI, Carnegie Mellon University
      avg_robot_linear_speed: true
      avg_robot_angular_speed: true
      avg_acceleration: true
      avg_overacceleration: true
      
      # Learning a Group-Aware Policy for Robot Navigation
      # Kapil Katyal ∗1,2 , Yuxiang Gao ∗2 , Jared Markowitz 1 , Sara Pohland 3 , Corban Rivera 1 , I-Jeng Wang 1 , Chien-Ming Huang 2 
      avg_pedestrian_velocity: true

      # metrics based on Social Force Model employed in 
      # different papers
      social_force_on_agents: true
      social_force_on_robot: true
      # social work function employed in this planner:
      # https://github.com/robotics-upo/social_force_window_planner
      social_work: true
      obstacle_force_on_robot: true
      obstacle_force_on_agents: true
    

